tensorboard/plugins/interactive_inference/witwidget/pip_package/setup.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from setuptools import find_packages, setup
+
+
+REQUIRED_PACKAGES = [
+    'ipywidgets>=7.0.0',
+    'jupyter>=1.0,<2',
+    'tensorflow>=1.12.0',
```

Comment:
FWIW - TensorBoard doesn't express a direct dep on TF because if this dep won't be satisfied by e.g. the `tensorflow-gpu` or `tf-nightly` packages, as pip isn't smart enough to realize that they're all providing the `tensorflow` package.  If the user already has one of those packages installed, then installing `witwidget` will cause `tensorflow` to be installed and clobber the existing package, which can result in a bit of a mess.

I guess tensorflow-serving-api already depends on TF directly, though (it partially gets around this by shipping a `-gpu` version that depends on `tensorflow-gpu` instead, apparently), so you're kind of stuck.  Just something to be aware of at least.

Reply:
hmm not ideal. thanks for explaining this.